src/collectors/hadoop/hadoop.py

```diff
@@ -101,6 +101,10 @@ def collect_from(self, filename):
                             data['name'],
                             metric, ]))
 
+                    if not self.path_allowed(path):
+                        # whitelist / blacklist
+                        continue
+
                     value = float(metrics[metric])
 
                     self.publish_metric(Metric(path,
```

Comment:
So I was trying to keep the issues and pull requests isolated, but there is overlap and so it is somewhat confusing. In my PR for multi black/white lists #245 I did move it to the publish method. Also renamed it metric_allowed.

I can combine these two things (path vs name & single vs multiple white/black lists) into one PR. Whatever you think makes more sense.

I'm going to rebase my existing PRs with the latest now...